homestead.md

```diff
@@ -514,6 +524,72 @@ Homestead uses the Nginx web server by default. However, it can install Apache i
 
 Homestead includes the Postfix mail transfer agent, which is listening on port `1025` by default. So, you may instruct your application to use the `smtp` mail driver on `localhost` port `1025`. Then, all sent mail will be handled by Postfix and caught by Mailhog. To view your sent emails, open [http://localhost:8025](http://localhost:8025) in your web browser.
 
+<a name="database-snapshots"></a>
+### Database Snapshots
+
+Homestead supports freezing the state of MySQL and MariaDB databases and branching between them with [Logical MySQL Manager](https://github.com/Lullabot/lmm). For example, imagine working on a site with a multi-gigabyte database. You can import the database and take a snapshot. After doing some work and creating some test content locally, to roll back to a known-good state you can quickly restore back to the original state. Under the hood, LMM uses LVM's thin snapshot functionality with copy-on-write support. In practice, this means that changing a single row in a table will only cause the changes you made to be written to disk, saving significant time and disk space during restores.
+
+Since `lmm` interacts with LVM, it must be run as root. To see all available commands, run `sudo lmm`.
+
+Note that there is no de-duplication process between snapshots. For example, if you create a snapshot with `sudo lmm branch production-2019-03-20`, drop all tables, and re-import the source database, you will have two completely independent sets of data on disk. To conserve disk space, consider importing one database and then running updates and migrations on top of it, and be sure to delete old snapshots with `sudo lmm delete`.
+
+<a name="debugging-and-profiling"></a>
+## Debugging & Profiling
+
+<a name="debugging-web-requests"></a>
+### Debugging Web Requests With Xdebug
+
+Homestead includes out-of-the-box support for step debugging with [Xdebug](https://xdebug.org). For example, you can load a web page from a browser, and PHP will connect back out to your IDE to allow inspection and modification of the running code.
+
+To enable debugging, run:
+
+    $ sudo phpenmod xdebug
+    $ sudo systemctl restart php7.3-fpm # Replacing the version with whatever PHP version are using.
+
+Then, follow your IDE's instructions to enable debugging. Finally, configure your browser to trigger Xdebug with an extension or [a bookmarklet](https://www.jetbrains.com/phpstorm/marklets/).
+
+Loading Xdebug by itself, even when not actively debugging, can significantly slow PHP. To disable Xdebug, run `sudo phpdismod xdebug` and restart the FPM service again.
+
+While Xdebug is included for all PHP versions, note that early minor releases of PHP often have some incompatibilities with Xdebug immediately after release. If PHP segfaults or throws Zend errors, try switching to the previous stable PHP release.
+
+<a name="debugging-cli-scripts"></a>
+### Debugging CLI scripts
+
+To debug a PHP CLI program, use the `xphp` shell alias:
+
+    $ xphp path/to/script
```

Comment:
Where?

Reply:
We should probably change this to `xphp path/to/script.php` or similar to give more context